### ISSUE TITLE
@import multiline support

### ISSUE THREAD

**Issue by dmitri-zganiaiko** [DannyBen/kojo#40]
Hi! Currently, in case of multiple arguments one has to write an `@import` in a long single line like this:

```yml
@import imported_file_name (argument1: 'long value 1', argument2: "long value 2", argument3: "long value 3", argument4: "long value 4")
```

What do you think about adding multiline support for `@import`?

```yml
@import imported_file_name ( \
argument1: 'long value 1', \
argument2: "long value 2", \
argument3: "long value 3", \
argument4: "long value 4" \
)
```

Here's working proof of concept with the required modification of `template.rb`:

```ruby
    def eval_imports(content)
      result = []
      multiline = nil

      content.lines.each do |line|
        line.chomp!
        spaces = line[/^\s*/].size

        if multiline.nil?
          if line =~ /^\s*@import .*\\$/
            multiline = line.delete_suffix('\\')
            next
          end
        else
          if line.end_with?("\\")
            multiline += line.delete_suffix("\\")
            next
          elsif line.end_with?(")")
            multiline += line
            line = multiline
            multiline = nil
          else
            raise Kojo::TemplateError, "Line should end either with ')' or '\': #{line}"
          end
        end

        if line =~ /^\s*@import ([^(\s]*)\s*(?:\((.*)\))?\s*/
          file = $1
          args = $2 || ''
          args = instance_eval("{#{args}}")
          imported = import file, args
          line = indent imported, spaces
        end

        result.push line
      end

      result.join "\n"
    end
```

**Comment by DannyBen** [comment]
Well, my initial reaction is this:

1. If you have that many arguments, and especially if they are long - you should probably look for ways to simplify your templates structure.
2. If we are to implement multiline support, then it should probably be a pre-processor one liner that straightens up the multiline into a single line, instead of complicating the each_line loop
3. Any multiline escape sequence - such as a `\` at the end of the line - needs to take into account that in some cases, people may use it and it should stay untouched (in any other context outside of the `@import` example).

While I am not rejecting the idea, at face value, I think this has the potential to cause more harm than good.

Personally, I am using Kojo with a very complex templates tree for Kuberenetes, and never had to use more than one or two arguments. 

Also - perhaps to mention that arguments are cascading down without the need to restate them:

### Input

```bash
# test.txt
before
@import imported (arg1: "value1")
after

# imported.txt
-- IMPORTED --
@import deeply_imported (arg2: "value2")
-- IMPORTED --

# deeply_imported.txt
got both %{arg1} and %{arg2}
```

### Output

```bash
$ kojo file test.txt

before
-- IMPORTED --
got both value1 and value2
-- IMPORTED --
after
```

**Comment by DannyBen** [comment]
I have an idea that might work. Let me tinker with it for a bit.

**Comment by DannyBen** [comment]
This is released in version 0.4.0.
